crates/sui-sdk-types/src/types/object.rs

```diff
@@ -322,6 +324,14 @@ impl Object {
     pub fn storage_rebate(&self) -> u64 {
         self.storage_rebate
     }
+
+    /// Convert this object into an [`unresolved::Input`] with the object id, digest, and version.
+    #[cfg(all(feature = "serde", feature = "hash"))]
+    pub fn as_input(&self) -> Input {
+        Input::by_id(self.object_id())
```

Comment:
```suggestion
    pub fn as_input(&self) -> super::unresolved::Input {
        super::unresolved::Input::by_id(self.object_id())
```

Reply:
I'm not actually a huge fan of this approach. honestly this should be handled by the TxnBuilder and not done here. I know there was some reasons with implementing `From` but maybe we end up moving the unresolved types out at some point.

This is just a comment on my preferences for having a cleaner api and something we can iterate on later (doesn't need to be fix now)